pkgs/by-name/on/onedrivegui/package.nix

```diff
@@ -0,0 +1,70 @@
+{ lib
+, python3Packages
+, fetchFromGitHub
+, copyDesktopItems
+, onedrive
+, makeDesktopItem
+}:
+
+python3Packages.buildPythonApplication rec {
+  pname = "OneDriveGUI";
+  version = "1.0.3";
+
+  src = fetchFromGitHub {
+    owner = "bpozdena";
+    repo = "OneDriveGUI";
+    rev = "v${version}";
+    hash = "sha256-HutziAzhIDYP8upNPieL2GNrxPBHUCVs09FFxdSqeBs=";
+  };
+
+  nativeBuildInputs = [ copyDesktopItems ];
+
+  propagatedBuildInputs = [ onedrive ] ++
+  (with python3Packages; [
+    pyside6
+    requests
+  ]);
+
+  doCheck = false; # No tests defined
+
+  desktopItems = [
+    (makeDesktopItem {
+      name = "OneDriveGUI";
+      exec = "onedrivegui";
+      desktopName = "OneDriveGUI";
+      comment = "OneDrive GUI Client";
+      type = "Application";
+      icon = "OneDriveGUI";
+      terminal = false;
+      categories = [ "Utility" ];
+    })
+  ];
+
+    preBuild = ''
+    cat > setup.py << EOF
+    from setuptools import setup
+    setup(
+      name='onedrivegui',
+      version='$version',
+      scripts=[
+        'src/OneDriveGUI.py',
+      ],
+    )
+    EOF
+  '';
+
+  postInstall = ''
+    mkdir $out/lib/OneDriveGUI && cp -r src/{resources,ui} $out/lib/OneDriveGUI/
```

Comment:
Usually it's a better idea to only use link (ln) from the $src, this way you're not duplicating the files and you're leaving Nix the handling task.